engine/access/state_stream/backend.go

```diff
@@ -67,15 +69,28 @@ func New(
 ) (*StateStreamBackend, error) {
 	logger := log.With().Str("module", "state_stream_api").Logger()
 
+	// cache the root block height and ID for runtime lookups.
+	rootHeight, err := state.Params().SporkRootBlockHeight()
```

Comment:
```suggestion
	rootHeight, err := state.Params().Root()
```

I would suggest to use `Root` instead of `SporkRootBlockHeight`, they are usually the same, except for dynamically bootstrapped access node. In that case, the node might not have blocks from `[SporkRootBlockHeight, RootHeight - 1]`, and also won't have execution data for it. 

Since blocks from `Root` height always exist, regardless is dynamically bootstrapped or not. So the AN will always download the execution data and make it available. 

cc @jordanschalm to confirm about the dynamic bootstrap 

If you agree, then the "spork root block" error message should be updated as well.

Reply:
Yes, nodes which join mid-spork will not have blocks between spork root and snapshot root.

Reply:
ah, yes. you're right. fixed here [2dbea0b](https://github.com/onflow/flow-go/pull/4271/commits/2dbea0bbc58eeaad5f9dfbc9f19cf9dbf4f1e8ec)